### ISSUE TITLE
Potential dependency conflicts between ablog and alabaster

### ISSUE THREAD

**Issue by NeolithEra** [sunpy/ablog#64]
Hi, as shown in the following full dependency graph of  **_ablog_**, **_ablog_** requires _**alabaster**_ (the latest version)， while the installed version of **_sphinx_**(3.0.1) requires _**alabaster>=0.7,<0.8**_.

According to Pip's “first found wins” installation strategy, _**alabaster 0.7.12**_ is the actually installed version.  

Although the first found package version _**alabaster 0.7.12**_ just satisfies the later dependency constraint （_**alabaster>=0.7,<0.8**_), it will lead to a build failure once developers release a newer version of alabaster.

### Dependency tree--------
```
ablog - 0.10.5
| +- alabaster(install version:0.7.12 version range:*)
| +- invoke(install version:1.4.1 version range:*)
| +- python-dateutil(install version:2.8.1 version range:*)
| +- sphinx(install version:3.0.1 version range:>=1.6)
| | +- alabaster(install version:0.7.12 version range:>=0.7,<0.8)
| | +- babel(install version:2.8.0 version range:>=1.3)
| | | +- pytz(install version:2019.3 version range:>=2015.7)
| | +- docutils(install version:0.16 version range:>=0.12)
| | +- imagesize(install version:1.2.0 version range:*)
| | +- Jinja2(install version:2.11.2 version range:>=2.3)
| | | +- MarkupSafe(install version:2.0.0a1 version range:>=0.23)
| | +- packaging(install version:20.3 version range:*)
| | +- Pygments(install version:2.6.1 version range:>=2.0)
| | +- requests(install version:2.23.0 version range:>=2.5.0)
| | | +- certifi(install version:2020.4.5.1 version range:>=2017.4.17)
| | | +- chardet(install version:3.0.4 version range:>=3.0.2,<4)
| | | +- idna(install version:2.9 version range:>=2.5,<3)
| | | +- urllib3(install version:1.25.9 version range:>=1.21.1,<1.26)
| | +- setuptools(install version:46.1.3 version range:*)
| | +- snowballstemmer(install version:2.0.0 version range:>=1.1)
| | +- sphinxcontrib-applehelp(install version:1.0.2 version range:*)
| | +- sphinxcontrib-devhelp(install version:1.0.2 version range:*)
| | +- sphinxcontrib-htmlhelp(install version:1.0.3 version range:*)
| | +- sphinxcontrib-jsmath(install version:1.0.1 version range:*)
| | +- sphinxcontrib-qthelp(install version:1.0.3 version range:*)
| | +- sphinxcontrib-serializinghtml(install version:1.1.4 version range:*)
| +- sphinx-automodapi(install version:0.12 version range:*)
| +- werkzeug(install version:1.0.1 version range:*)
```

Thanks for your attention.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution
1.	Loosen the version range of **_alabaster_** to be **_<=0.7.12_**.
2.	Remove your direct dependency **_alabaster_**, and use the **_alabaster_** transitively introduced by **_sphinx_**.
3.	Change your direct dependency **_sphinx_** to be **_<=3.0.1,>=1.6_**.

@lordnapi Which solution do you prefer, 1,2 or 3?
Please let me know your choice. May I pull a request to solve this issue?

**Comment by nabobalis** [comment]
Hi @NeolithEra, thanks for the issue but I was under the impression ablog does not pin its version of alabaster, we have had to fix werkzeug<1 and  nbsphinx<0.5.0. 

So I am unsure about this problem.

**Comment by nabobalis** [comment]
I will close this for the time being, if you want to suggest a fix, I would strongly suggest opening a pull request.
